### DISCUSSION TITLE
Label outside shape should not count for the shapes size

### DISCUSSION THREAD

**Discussion by TobiasRama** [clientIO/joint#1915] [Q&A] [answered]
I've created a shape with a label which is next to it. This shape has a custom button in its upper right corner. After adding the label, the size of the shape has changed, so that the button isn't longer in the corner but next to the label. Is it possible to ignore the size of the label, so that it doesn't count for the size of the shape? Another idea would be to set the position of the button relative to the rect element but I don't know how to get the width and height of the rect.
This is how it looks:
![image](https://user-images.githubusercontent.com/59836233/204502457-1aa13fbb-cd4b-4b08-8bbd-74452d937d43.png)
This is how it should look like:
![image](https://user-images.githubusercontent.com/59836233/204502607-deab94a0-1c48-4885-9982-1019229444d5.png)

**Comment by kumilingus** [comment, marked as answer]
Use `useModelGeometry: true` option.

```js
const infoButton = new joint.elementTools.Button({
    focusOpacity: 0.5,
    // top-right corner
    x: '100%',
    y: '0%',
    useModelGeometry: true, 
    action: function(evt) {
        alert('View id: ' + this.id + '\n' + 'Model id: ' + this.model.id);
    },
    markup: [{
        tagName: 'circle',
        selector: 'button',
        attributes: {
            'r': 7,
            'fill': '#001DFF',
            'cursor': 'pointer'
        }
    }, {
        tagName: 'path',
        selector: 'icon',
        attributes: {
            'd': 'M -2 4 2 4 M 0 3 0 0 M -2 -1 1 -1 M -1 -4 1 -4',
            'fill': 'none',
            'stroke': '#FFFFFF',
            'stroke-width': 2,
            'pointer-events': 'none'
        }
    }]
});
```

It seems to be missing from the documentation.

**Reply by TobiasRama** [reply]
Thank for your reply, it solved my problem. But I have another question. I've set up a custom router which calculate points with the bboxes of source and target. My problem is, that the arrowheads do not point at the middle of the rect but on the middle of the whole shape. 

![image](https://user-images.githubusercontent.com/59836233/204518212-382b242b-381a-4b97-aacf-c61301888daf.png)
Here is an example of how I a calculate the first point.
```

defaultRouter: (vertices: any, opt: any, linkView: any) => {

                const sv = linkView.getEndView('source');
                const tv = linkView.getEndView('target');

                const sourceBBox = sv.model.getBBox();

                const length = 25;
                const p1 = {
                    x: sourceBBox.x + sourceBBox.width / 2,
                    y: sourceBBox.y + sourceBBox.height + length
                };
}
```

**Reply by kumilingus** [reply]
It's the same issue as https://github.com/clientIO/joint/discussions/1860.

**Reply by kumilingus** [reply]
Using the [modelCenter](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#anchors.modelCenter) anchor should also work.
